core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -278,4 +278,21 @@ package object config {
         "spark.io.compression.codec.")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val SHUFFLE_ACCURATE_BLOCK_THRESHOLD =
+    ConfigBuilder("spark.shuffle.accurateBlkThreshold")
+      .doc("When we compress the size of shuffle blocks in HighlyCompressedMapStatus, we will " +
+        "record the size accurately if it's above the threshold specified by this config. This " +
+        "helps to prevent OOM by avoiding underestimating shuffle block size when fetch shuffle " +
+        "blocks.")
+      .longConf
+      .createWithDefault(100 * 1024 * 1024)
+
+  private[spark] val REDUCER_MAX_REQ_SIZE_SHUFFLE_TO_MEM =
+    ConfigBuilder("spark.reducer.maxReqSizeShuffleToMem")
```

Comment:
If we want to avoid fetch big requests into memory, shall we simply do this for all blocks sized over `SHUFFLE_ACCURATE_BLOCK_THRESHOLD`? Instead of going to compare with `REDUCER_MAX_REQ_SIZE_SHUFFLE_TO_MEM` at fetching time?

To have two configs for the one final goal seems to cause more complexity.